### ISSUE TITLE
Garage Door support

### ISSUE THREAD

**Issue by Faisalthe01** [basdelfos/homebridge-tuya-web#80]
i bought this garage door and added it into Smart Life.

would it be possible to add it in Homebridge as a garage door with open/close sensor support

https://www.ebay.com.au/itm/USB-Powered-Smart-Garage-Door-Gate-Controller-Opener-APP-WiFi-Remote-Control-AU/193217731305?ssPageName=STRK%3AMEBIDX%3AIT&_trksid=p2057872.m2749.l2649

**Comment by vitramir** [comment]
It is possible, but we need more information about device. First of all you should get warning "Could not init class for device type <DEVICE TYPE HERE>". Provide us with device type.

**Comment by Faisalthe01** [comment]
it does not even find it, it can only find scene that I created in Smart Life app for it

[4/1/2020, 11:32:36 PM] [TuyaWebPlatform] Adding:  "Garage " switch:OFF (scene / puyAymJeNMExLUYd)
[4/1/2020, 11:32:36 PM] [TuyaWebPlatform] Could not init class for device type [scene]
[4/1/2020, 11:32:36 PM] [TuyaWebPlatform] Adding:  "Garage " switch:ON (scene / pMuZiHnHj91A6hAY)
[4/1/2020, 11:32:36 PM] [TuyaWebPlatform] Could not init class for device type [scene]

**Comment by dewoert** [comment]
Same request from me.
How can we help to get a working Garage Door Opener ?
I tried to find the IDs, but no luck.
Only have a Virtual ID (found in TuyaSmart App) ?

**Comment by dewoert** [comment]
Perhaps a little bit more info:

```
[TuyaAccessory] Raw message from Garage Door Opener (3.3) with command 8: 000055aa00000000000000080000006b00000000332e330000000000000052000000016545b76be389dd922e2c646578d861675224af5c9633b2735585a001bb0395a89edd79c60a624d479910d14eb959dd7685f0c9b1adb0bb37dc5d62c7bf1b49b17ce20d33ce05ee9e56c1930a0b9f1bcc43fedc1f0000aa55

[TuyaAccessory] Odd message from Garage Door Opener with command 8: eE�k�ݒ.,dex�agR$�\�3�sU�����$r�ߩ�{.��
                                                                                                         �e*Ă�������윓n�

[TuyaAccessory] Raw message from Garage Door Opener (3.3) with command 8: 000055aa00000000000000080000005b00000000332e330000000000000053000000016545b76be389dd922e2c646578d861675224af5c9633b2735585a001bb0395a82472ccdfa9ba7b2ee7e50be2652ac482baf7a41b4989ccf7029912ec9c936ed22f93289a0000aa55

[TuyaAccessory] Odd message from Garage Door Opener with command 8: eE�k�ݒ.,dex�agR$�\�3�sU�����$r�ߩ�{.��
                                                                                                         �e*Ă�������윓n�

[TuyaAccessory] Raw message from Garage Door Opener (3.3) with command 8: 000055aa00000000000000080000005b00000000332e330000000000000054000000016545b76be389dd922e2c646578d861675224af5c9633b2735585a001bb0395a82472ccdfa9ba7b2ee7e50be2652ac482baf7a41b4989ccf7029912ec9c936ed2691a572d0000aa55

```

**Comment by KevinDevs** [comment]
[TuyaWebPlatform] Adding: Garage door (switch / XXX)
[TuyaWebPlatform] Creating New Accessory XYZ
[TuyaWebPlatform] Adding: Close Garage Door (scene / XZY)
[TuyaWebPlatform] Could not init class for device type [scene]
[TuyaWebPlatform] Adding: Open Garage Door (scene / YZX)
[TuyaWebPlatform] Could not init class for device type [scene]

**Comment by Faisalthe01** [comment]
These logs are for scenes and not for the accessory. Don’t think Tuya Web even picks it up

**Comment by KevinDevs** [comment]
Thats the only thing I'm getting though..
Right now, it detects my garage door as a outlet though..

**Comment by Faisalthe01** [comment]
Oh you have that Opener, i had that one before i returned it coz it was showing as outlet in Google Home. The new one shows as door in Google Home but Tuya Web don’t even pick it up. Since your one is showing in Homebridge, maybe someone can write a code for a garage door in Tuya Web and then you can change the accessory to garage Door in your config.

**Comment by KevinDevs** [comment]
Yes, Mine is showing as a door too in google home, but ain't gonna blame nothing since I got it for free lol. Hopefully there might be a solution for this that makes it compatible for home bridge
